library/init/data/nat/gcd.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2014 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Mario Carneiro
+
+Definitions and properties of gcd, lcm, and coprime.
+-/
+prelude
+import init.data.nat.lemmas init.wf
+
+open well_founded
+
+namespace nat
+
+/- gcd -/
+
+def gcd.F : Π (y : ℕ), (Π (y' : ℕ), y' < y → nat → nat) → nat → nat
```

Comment:
Yes. We should probably also define `div` and `mod` using the equation compiler.